pkg/controller/gardenletdeployer/actuator_test.go

```diff
@@ -529,7 +533,73 @@ var _ = Describe("Interface", func() {
 				))
 			})
 
+			It("should return error when DoNotCopyBackupCredentials feature gate is enabled and backup secret does not exist", func() {
+				DeferCleanup(test.WithFeatureGate(features.DefaultFeatureGate, features.DoNotCopyBackupCredentials, true))
+
+				expectCheckSeedSpec()
+				recorder.EXPECT().Eventf(managedSeed, corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, "Ensuring garden namespace in target cluster")
+				expectCreateGardenNamespace()
+				recorder.EXPECT().Event(managedSeed, corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, "Reconciling seed secrets")
+
+				gardenClient.EXPECT().Get(ctx, client.ObjectKey{Namespace: namespace, Name: backupSecretName}, gomock.AssignableToTypeOf(&corev1.Secret{})).DoAndReturn(
+					func(_ context.Context, _ client.ObjectKey, _ *corev1.Secret, _ ...client.GetOption) error {
+						return apierrors.NewNotFound(corev1.Resource("secret"), backupSecretName)
+					},
+				)
+
+				recorder.EXPECT().Eventf(managedSeed, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, gomock.Any())
+
+				_, err := actuator.Reconcile(ctx, log, managedSeed, managedSeed.Status.Conditions, managedSeed.Spec.Gardenlet.Deployment, &gardenlet.Config, *managedSeed.Spec.Gardenlet.Bootstrap, *managedSeed.Spec.Gardenlet.MergeWithParent)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("could not reconcile seed test secrets: the configured backup secret does not exist, however the feature gate DoNotCopyBackupCredentials is enabled and shoot infrastructure credentials will not be reused"))
+			})
+
+			It("should remove owner reference from backup secret when DoNotCopyBackupCredentials feature gate is enabled", func() {
```

Comment:
Does it make sense to extract the 2 test for `DoNotCopyBackupCredentials=true` to a When section

```go
			When("DoNotCopyBackupCredentials feature gate is enabled", func() {
				BeforeEach(func() {
					DeferCleanup(test.WithFeatureGate(features.DefaultFeatureGate, features.DoNotCopyBackupCredentials, true))
				})

				It("should return error when backup secret does not exist", func() {

				})

				It("should remove owner reference from backup secret when backup secret exists", func() {

				})
			})
```

WDYT?